src/vcpkg/base/files.cpp

```diff
@@ -2031,14 +2031,15 @@ namespace vcpkg
     {
         this->rename(old_path, new_path, ec);
         using namespace std::chrono_literals;
+        std::error_code local_ec;
         for (const auto& delay : {10ms, 100ms, 1000ms, 10000ms})
         {
             if (!ec)
             {
                 return true;
             }
             else if (ec == std::make_error_condition(std::errc::directory_not_empty) ||
-                     ec == std::make_error_condition(std::errc::file_exists) || this->exists(new_path, ec))
+                     ec == std::make_error_condition(std::errc::file_exists) || this->exists(new_path, local_ec))
```

Comment:
Overwriting `ec` here broke early `return false` for `std::errc::cross_device_link`.

Reply:
BTW the forced formatting didn't let me put the trailing `this->exists` on a new line were it would be much more obvious...